src/pages/how-does-react-tell-a-class-from-a-function.ja.md

```diff
@@ -0,0 +1,557 @@
+---
+title: Reactはどうやって関数からクラスを見分けるているの？
+date: '2018-12-02'
+langs: ['en', 'ja']
+spoiler: クラス、new、instanceof、プロトタイプチェーン、およびAPI設計について説明します。
+---
+
+関数として定義された`Greeting`コンポーネントについて考えてみましょう:
+
+```jsx
+function Greeting() {
```

Comment:
You can translate these, too if you’d like — A quick test shows that Japanese characters are valid function names.

Reply:
Not sure what the conventions are for writing code in languages other than English, though.

Reply:
I think for consistency I'd leave code snippets original (except comments)